### PR TITLE
Fix for NRE in VSO Watson bug 217660

### PR DESCRIPTION
**Pull request by mavasani**
Add null checks before raising events.

Fixes [217660](https://devdiv.visualstudio.com/DevDiv/_workitems?_a=edit&id=217660&triage=true)

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @dotnet/roslyn-ide @dotnet/roslyn-analysis for review.

**Comment by srivatsn** [comment]
👍

File: src/Workspaces/Core/Portable/Workspace/DocumentEventArgs.cs [hunk 1]
```diff
@@ -10,6 +11,7 @@ public class DocumentEventArgs : EventArgs
 
         public DocumentEventArgs(Document document)
         {
+            Contract.ThrowIfNull(document);
```
Comment (jmarolf) [inline src/Workspaces/Core/Portable/Workspace/DocumentEventArgs.cs]: Why are we being passed a null document?

Comment (mavasani) [inline src/Workspaces/Core/Portable/Workspace/DocumentEventArgs.cs]: [This](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Workspace_Editor.cs,364) can pass in a null argument for a removed document.

**Comment by heejaechang** [comment]
is this from custom workspace?

**Comment by heejaechang** [comment]
ah, alright +1

**Comment by jmarolf** [comment]
test vsi please

**Comment by jmarolf** [comment]
:+1:

**Pull request merged**